app/src/main/java/pl/futurecollars/invoicing/db/FileDatabase.java

```diff
@@ -46,37 +46,42 @@ public List<Invoice> getAll() {
   }
 
   @Override
-  public void update(long id, Invoice updatedInvoice) {
-
+  public Optional<Invoice> update(long id, Invoice updatedInvoice) {
     List<Invoice> invoicesList = getAll();
-    invoicesList
-        .stream()
-        .filter(invoice -> invoice.getId().equals(id))
-        .findFirst()
-        .ifPresentOrElse(invoice -> {
-          invoice.setData(updatedInvoice.getData());
-          invoice.setBuyer(updatedInvoice.getBuyer());
-          invoice.setSeller(updatedInvoice.getSeller());
-        },
-            () -> {
-              throw new IllegalArgumentException("Faktura o numerze: " + id + " nie istnieje");
-            });
-
+    try {
+      invoicesList
+          .stream()
+          .filter(invoice -> invoice.getId().equals(id))
+          .findFirst()
+          .ifPresentOrElse(invoice -> {
+            invoice.setData(updatedInvoice.getData());
+            invoice.setBuyer(updatedInvoice.getBuyer());
+            invoice.setSeller(updatedInvoice.getSeller());
+          },
+              () -> {
+                throw new IllegalArgumentException("Faktura o numerze: " + id + " nie istnieje");
```

Comment:
Niepotrzebnie rzucasz wyjątek, by za moment go złapać. Obsługa wyjątków jest "ciężka" dla programu, więc jeśli można jej uniknąć (a można) to piszmy kod tak, aby tego unikać. Wystarczy, że skorzystasz z innej metody niż ifPresentOrElse.

Reply:
poprawki zrobione.